enhancements/installer/gcp-customer-provided-subnets.md

```diff
@@ -0,0 +1,258 @@
+  Options: results of a [subnetworks.list][subnetworks-list] call filtered by the selected VPC.
+  Only when: VPC is not “new”
+
+#### Isolation between clusters
+
+The clusters will be isolated by firewall rules. Specifically, by tagging the instances with ${INFRA_ID}-master and ${INFRA_ID}-worker, and using those tags in the firewall rules to ensure only desired traffic is allowed. However, there are a few modifications required to make this absolute:
+
+- ICMP ingress.  We may be able to drop this completely.
+
+- TCP 22 ingress (SSH). The bootstrap instance  allows global access (0.0.0.0/0). This can probably stay the same as it is only temporary and of limited risk.  The control-plane and compute instances allow the network_cidr [Networking.MachineCIDR][networking-machinecidr]. This could be restricted to tags if determined the risk is sufficient. Using different ssh keys would help as well.
+
+- TCP 22623 ingress (MCS). The MCS firewall rule is to be opened up externally in order to remove the need for master_nat_ip and worker_nat_ip as inputs. Switching to using internal load balancers would remediate this.
+
+#### Destroy
+
+There should be no work required. For gcp, we filter on name containing {INFRA_ID}. As a result, destroy should only delete things that match this naming convention.
+
+#### CI
+
+CI jobs should set up a VPC, subnets, routing, and NAT using our [UPI flow][upi-flow] and launch the cluster-under-test into that infrastructure.
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.
+
+How will security be reviewed and by whom? How will UX be reviewed and by whom?
+
+Consider including folks that also work outside your immediate sub-project.
+
+## Design Details
+
+### Test Plan
+
+To test multiple clusters can be created in same networking deployment, CI account will host a networking deployment and the CI templates will be modified to use the subnets to create a cluster.
+
+### Graduation Criteria
+
+*This enhancement will follow standard graduation criteria.
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+### Upgrade / Downgrade Strategy
+
+Not applicable
+
+### Version Skew Strategy
+
+Not applicable
+
+## Implementation History
+
+## Drawbacks
+
+Customer owned networking components means the cluster cannot automatically change things about the networking during upgrade.
+
+## Alternatives
+
+Similar to the `Drawbacks` section the `Alternatives` section is used to
```

Comment:
more boilerplate that you should replace with `Not applicable.` or some set of considered alternatives.